compiler_opt/es/policy_utils_test.py

```diff
@@ -0,0 +1,210 @@
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+###############################################################################
+#
+#
+# This is a port of the code by Krzysztof Choromanski, Deepali Jain and Vikas
+# Sindhwani, based on the portfolio of Blackbox optimization algorithms listed
+# below:
+#
+# "On Blackbox Backpropagation and Jacobian Sensing"; K. Choromanski,
+#  V. Sindhwani, NeurIPS 2017
+# "Optimizing Simulations with Noise-Tolerant Structured Exploration"; K.
+#  Choromanski, A. Iscen, V. Sindhwani, J. Tan, E. Coumans, ICRA 2018
+# "Structured Evolution with Compact Architectures for Scalable Policy
+#  Optimization"; K. Choromanski, M. Rowland, V. Sindhwani, R. Turner, A.
+#  Weller, ICML 2018, https://arxiv.org/abs/1804.02395
+#  "From Complexity to Simplicity: Adaptive ES-Active Subspaces for Blackbox
+#   Optimization";  K. Choromanski, A. Pacchiano, J. Parker-Holder, Y. Tang, V.
+#   Sindhwani, NeurIPS 2019
+# "i-Sim2Real: Reinforcement Learning on Robotic Policies in Tight Human-Robot
+#  Interaction Loops"; L. Graesser, D. D'Ambrosio, A. Singh, A. Bewley, D. Jain,
+#  K. Choromanski, P. Sanketi , CoRL 2022, https://arxiv.org/abs/2207.06572
+# "Agile Catching with Whole-Body MPC and Blackbox Policy Learning"; S.
+#  Abeyruwan, A. Bewley, N. Boffi, K. Choromanski, D. D'Ambrosio, D. Jain, P.
+#  Sanketi, A. Shankar, V. Sindhwani, S. Singh, J. Slotine, S. Tu, L4DC,
+#  https://arxiv.org/abs/2306.08205
+# "Robotic Table Tennis: A Case Study into a High Speed Learning System"; A.
+#  Bewley, A. Shankar, A. Iscen, A. Singh, C. Lynch, D. D'Ambrosio, D. Jain,
+#  E. Coumans, G. Versom, G. Kouretas, J. Abelian, J. Boyd, K. Oslund,
+#  K. Reymann, K. Choromanski, L. Graesser, M. Ahn, N. Jaitly, N. Lazic,
+#  P. Sanketi, P. Xu, P. Sermanet, R. Mahjourian, S. Abeyruwan, S. Kataoka,
+#  S. Moore, T. Nguyen, T. Ding, V. Sindhwani, V. Vanhoucke, W. Gao, Y. Kuang,
+#  to be presented at RSS 2023
+###############################################################################
+"""Tests for policy_utils."""
+
+from absl.testing import absltest
+import numpy as np
+import os
+import tensorflow as tf
+from tf_agents.networks import actor_distribution_network
+from tf_agents.policies import actor_policy
+
+from compiler_opt.es import policy_utils
+from compiler_opt.rl import policy_saver, registry
+from compiler_opt.rl.inlining import InliningConfig
+from compiler_opt.rl.inlining import config as inlining_config
+from compiler_opt.rl.regalloc import config as regalloc_config
+from compiler_opt.rl.regalloc import RegallocEvictionConfig, regalloc_network
+
+
+class ConfigTest(absltest.TestCase):
```

Comment:
Maybe add a "TODO" on this with an issue number to move these elsewhere - it's a bit odd that very problem-specific tests mix with very general-purpose utilities. We should tie the same issue # with how create_actor_policy could have less of a dependency on problem_config registry, and just be given the needed parameters - the signature spec and processing layer creator; then the unittest wouldn't need to depend on specific problem solvers.

There's still value in having the latter, just elsewhere.

Reply:
This is being tracked under Issue #280